source/extensions/filters/network/thrift_proxy/router/router_impl.h

```diff
@@ -22,8 +22,24 @@ namespace NetworkFilters {
 namespace ThriftProxy {
 namespace Router {
 
+/**
+ * Base interface for something that matches MessageMetadata.
+ */
+class Matchable {
```

Comment:
Do you have a plan for this? I don't think this change uses this interface anywhere? Currently we pass around RouteEntryImplBase and just expect it'll have a matches method.

Reply:
I don't, actually. I started out cribbing [config_impl.cc](https://github.com/envoyproxy/envoy/blob/1381673ad2d74bab4667942abdd8ef75c812e75e/source/common/router/config_impl.h#L36) and had actually implemented it _without_ this interface but added it back in case there was something I was missing. Seems like it was just cruft after a second look. 

Should I just kill it? Would rather not bring over more cruft than necessary.